components/scream/src/physics/shoc/shoc_constants.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef SHOC_CONSTANTS_HPP
+#define SHOC_CONSTANTS_HPP
+
+#include "share/scream_types.hpp"
+
+#include <vector>
+
+namespace scream {
+namespace shoc {
+
+/*
+ * Mathematical constants used by SHOC.
+ *
+ * Note that a potential optimization could be to change the type of
+ * Scalar constants that have integer values to int.
+ */
+
+template <typename Scalar>
+struct Constants
+{
+  // These were taken from the PhysConst class in
+  // E3SM-Project/scream-docs/shoc-port/shocintr.py.
```

Comment:
Note the python file has the comment `# WARNING: Do not use these for anything real; use simulation-consistent values.` for these. You should get the true values by following the modules to where the E3SM constants live.

Reply:
If this were _only_ for the regression test, it would be OK to use these. But `physics/shoc/shoc_constants.hpp` implies these are real. Also, some if not all of these should go in a common constants file for scream. Finally, I believe eventually most of scream's constants should be init'ed by the F90 E3SM driver for consistency.

Reply:
Several of these values match those in p3_constants.hpp, so they're probably not total garbage. I was planning to consolidate all these constants into physics_constants.{hc}pp, but if E3SM is going to initialize these things, they're not "constants" in the software sense--they become regular global variables whose values are set on initialization and not afterward.

Is it worth addressing this E3SM initialization bit now that we're discussing it, or do you think we should kick the can down the road and get SHOC up and running? @ambrad @jgfouca

Reply:
Kick the can down the road. Consolidation is sufficient for now.